File: m/rls/rlb/stlst.c
/* file -  stlst - last character */
#include "m:\rid\rider.h"
#include "m:\rid\stdef.h"
/* code -  st_lst - look at last character */
char *st_lst(
register char *str )
{ if( *str == 0)return ( str );
  while ( *str++) {;}
  return ( str - 2);
} 

File: m/rls/stlst.r
file	stlst - last character
include rid:rider
include rid:stdef

code	st_lst - look at last character

;	st_lst : (str : *char) => *char
;
;	reply str if *str eq		; null case
;	reply st_end (str) - 1		; reply last
;
;	if str[0]			; null case
;	   while str[1]			; string end
;	.. .. ++str			;
;	reply str			; reply last

  func	st_lst
	str : * char~			; string
	()  : * char			; at last character
  is	reply str if *str eq		; null string
	spin while *str++		; get to the end quickly
	reply str - 2			; backup to last character
  end
